src/main/java/ch/njol/skript/lang/Expression.java

```diff
@@ -247,16 +250,16 @@ default boolean canReturn(Class<?> returnType) {
 	Expression<?> getSource();
 
 	/**
-	 * Simplifies the expression, e.g. if it only contains literals the expression may be simplified to a literal, and wrapped expressions are unwrapped.
-	 * <p>
-	 * After this method was used the toString methods are likely not useful anymore.
-	 * <p>
-	 * This method is not yet used but will be used to improve efficiency in the future.
+	 * Attempts to create a {@link SimplifiedLiteral} by evaluating the expression with a {@link ContextlessEvent}.
+	 * This should only be attempted IFF the expression's children are all literals and
+	 * {@link #getAll(Event)} would always return the exact same value, no matter the context in which it is called.
+	 * The value of {@link #toString(Event, boolean)} will be evaluated and captured for the literal's toString methods.
 	 *
-	 * @return A reference to a simpler version of this expression. Can change this expression directly and return itself if applicable, i.e. no references to the expression before
-	 *         this method call should be kept!
+	 * @return A simplified literal with the data from this expression's evaluation.
 	 */
-	Expression<? extends T> simplify();
+	default Literal<T> getAsSimplifiedLiteral() {
```

Comment:
I'm not sure if this method is really necessary. Assuming we have proper API documentation in the future, we might be best off to just direct users to call the SimplifiedLiteral method